src/main.rs

```diff
@@ -144,34 +144,32 @@ const FRONTARY_REPO_NAME: &str = "frontary";
 
 fn main() -> Result<(), io::Error> {
     let args = Args::parse();
-    let ui_local = args.ui_path.clone();
-    let fr_local = args.frontary_path.clone();
-    let ssh_key = args.ssh_key;
 
-    initialize_ssh(&ssh_key)?;
-    log_repo_strategy(ui_local.as_ref(), fr_local.as_ref());
+    validate_ssh_key_path(&args.ssh_key).map_err(|e| io::Error::other(e.message().to_owned()))?;
 
-    let repo_manager =
-        RepoManager::new().map_err(|e| io::Error::other(format!("Temp dir error: {e}")))?;
+    let repo_manager = RepoManager::new_with_key(args.ssh_key.clone())
+        .map_err(|e| io::Error::other(format!("Failed to create RepoManager: {e}")))?;
 
-    let ui_repo = prepare_repo(AICE_WEB_URL, ui_local, UI_REPO_NAME, &repo_manager)?;
-    let fr_repo = prepare_repo(FRONTARY_URL, fr_local, FRONTARY_REPO_NAME, &repo_manager)?;
+    log_repo_strategy(args.ui_path.as_ref(), args.frontary_path.as_ref());
 
-    checkout_frontary(
-        args.frontary_path.as_ref(),
-        ui_repo.as_path(),
-        fr_repo.as_path(),
+    println!("🛠️ Cloning repository: {AICE_WEB_URL}...");
```

Comment:
Currently, the log is printed unconditionally regardless of whether a clone actually happens. Could you please modify the code so that the log is only printed when a clone is performed inside the prepare_repo function?